### PR TITLE
Mike Hernandez

### PR DESCRIPTION
**Pull request by mikez321**
so close to making that inject/reduce method work!! was like 1 step away!

### CODE REVIEW

File: lib/show.rb [hunk 1]
```diff
@@ -0,0 +1,27 @@
+class Show
+  attr_reader :name, :creator, :characters
+  def initialize(name, creator, characters)
+    @name = name
+    @creator = creator
+    @characters = characters
+  end
+
+  def total_salary
+    total = characters.map do |character|
```
Comment (memcmahon) [inline lib/show.rb]: This `map` and then `sum` definitely works, but you could also use `sum` as an enumerable.

File: lib/show.rb [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+      character.info[:salary]
+    end
+    total.sum
+  end
+
+  def highest_paid_actor
+    pay_rates = characters.sort_by { |character| character.info[:salary] }.reverse
```
Comment (memcmahon) [inline lib/show.rb]: Good use of sort_by - you could also use the enumerable `max_by` here

File: lib/show.rb [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    pay_rates.first.info[:actor]
```
Comment (memcmahon) [inline lib/show.rb]: Again, it is better to use the `actor` method, rather than the actor key on the info hash.

File: lib/show.rb [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+  end
+
+  def actors
+    characters.map do |character|
```
Comment (memcmahon) [inline lib/show.rb]: Great `map`!

File: lib/network.rb [hunk 5]
```diff
@@ -1 +1,34 @@
...
+  def initialize(name)
+    @name = name
+    @shows = []
+  end
 
+  def add_show(show)
+    @shows << show
+  end
+
+  def main_characters
```
Comment (memcmahon) [inline lib/network.rb]: Main characters is a bit more involved than a list of all characters - keep an eye on the interaction pattern for explanations of what specific methods should be doing!  What we were really looking for are all characters who make more than 500,000 AND their character name is all caps.

File: lib/network.rb [hunk 6]
```diff
@@ -1 +1,34 @@
...
+    end.flatten
+  end
+
+  def actors_by_show
+    shows.reduce ({}) do |listing, shows|
+
+      if listing.include?(shows)
+        shows << shows.characters
+      else
+        listing[shows.name] = shows.characters
```
Comment (memcmahon) [inline lib/network.rb]: sooo close! on this line, you want to set the value to the array `[shows.characters]`; that way you can start shoveling into array.

File: test/network_test.rb [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+    michael_knight = Character.new({name: "Michael Knight", actor: "David Hasselhoff", salary: 1_600_000})
+    kitt = Character.new({name: "KITT", actor: "William Daniels", salary: 1_000_000})
+    knight_rider = Show.new("Knight Rider", "Glen Larson", [michael_knight, kitt])
+    leslie_knope = Character.new({name: "Leslie Knope", actor: "Amy Poehler", salary: 2_000_000})
+    ron_swanson = Character.new({name: "Ron Swanson", actor: "Nick Offerman", salary: 1_400_000})
+    parks_and_rec = Show.new("Parks and Recreation", "Michael Shur & Greg Daniels", [leslie_knope, ron_swanson])
+    nbc.add_show(knight_rider)
+    nbc.add_show(parks_and_rec)
+
+    assert_equal [michael_knight, kitt, leslie_knope, ron_swanson], nbc.main_characters
```
Comment (memcmahon) [inline test/network_test.rb]: Keep an eye on the interaction pattern!! This test is giving you a false positive - based on the interaction pattern, the only main character for these shows is `kitt`

File: test/network_test.rb [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+    kitt = Character.new({name: "KITT", actor: "William Daniels", salary: 1_000_000})
+    knight_rider = Show.new("Knight Rider", "Glen Larson", [michael_knight, kitt])
+    leslie_knope = Character.new({name: "Leslie Knope", actor: "Amy Poehler", salary: 2_000_000})
+    ron_swanson = Character.new({name: "Ron Swanson", actor: "Nick Offerman", salary: 1_400_000})
+    parks_and_rec = Show.new("Parks and Recreation", "Michael Shur & Greg Daniels", [leslie_knope, ron_swanson])
+    nbc.add_show(knight_rider)
+    nbc.add_show(parks_and_rec)
+    
+    expected ={
+      "Knight Rider" => ["David Hasselhoff", "William Daniels"],
```
Comment (memcmahon) [inline test/network_test.rb]: Watch the interaction pattern!  The keys in this hash should be show objects, not the name of a show.

File: test/show_test.rb [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+    kitt = Character.new({name: "KITT", actor: "William Daniels", salary: 1_000_000})
+    michael_knight = Character.new({name: "Michael Knight", actor: "David Hasselhoff", salary: 1_600_000})
+    knight_rider = Show.new("Knight Rider", "Glen Larson", [michael_knight, kitt])
+
+    assert_equal "Knight Rider", knight_rider.name
+    assert_equal "Glen Larson", knight_rider.creator
+    assert_equal [michael_knight, kitt], knight_rider.characters
+  end
+
+  def test_it_costs_a_lot_of_money
```
Comment (memcmahon) [inline test/show_test.rb]: 😆

File: lib/network.rb (hunk 5 above)
Comment (mikez321) [reply lib/network.rb]: I heard people asking about this but figured I wasn't there yet...

File: test/network_test.rb (hunk 8 above)
Comment (mikez321) [reply test/network_test.rb]: I had the right expected result at first, but after working for 40 something minutes trying to get reduce to work (it was my first time trying to use it!) I got something really close and got distracted at what I was supposed to be getting....

**Pull request closed without merging**